amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfigurator.java

```diff
@@ -69,7 +71,30 @@ public MultiLangDaemonConfiguration getConfiguration(Properties properties) {
         });
 
         Validate.notBlank(configuration.getApplicationName(), "Application name is required");
-        Validate.notBlank(configuration.getStreamName(), "Stream name is required");
+
+        if (configuration.getStreamArn() != null && !configuration.getStreamArn().trim().isEmpty()) {
+            Arn streamArnObj = Arn.fromString(configuration.getStreamArn());
+            if(!streamArnObj.getResource().getResourceType().equalsIgnoreCase("stream")){
+                throw new IllegalArgumentException(String.format("StreamArn has unsupported resource type of '%s'. Expected: stream",
+                        streamArnObj.getResource().getResourceType()));
+            }
+            if(!streamArnObj.getService().equalsIgnoreCase("kinesis")){
+                throw new IllegalArgumentException(String.format("StreamArn has unsupported service type of '%s'. Expected: kinesis",
+                        streamArnObj.getResource().getResourceType()));
+            }
+            //Parse out the stream Name from the Arn (and/or override existing value for Stream Name)
+            String streamNameFromArn = streamArnObj.getResource().getResource();
+            configuration.setStreamName(streamNameFromArn);
+
+            //Parse out the region from the Arn and set (and/or override existing value for region)
+            Region regionObj = Region.of(streamArnObj.getRegion());
+            if(Region.regions().stream().filter(x -> x.id().equalsIgnoreCase(regionObj.id())).count() == 0){
+                throw new IllegalArgumentException(String.format("%s is not a valid region", regionObj.id()));
+            }
+            configuration.setRegionName(regionObj);
```

Comment:
This may be incorrect.

For cross-account access, it's valid for the KCL application to execute in region `A`, the Kinesis endpoint to be in region `B`, and the Kinesis stream to be in region `C`. An ARN with region `C` should not invalidate or replace `B` or `A`.